streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -996,6 +1082,62 @@ private boolean close(final long timeoutMs) {
         }
     }
 
+    private void closeToError() {
+        if (!setState(State.ERROR)) {
+            log.info("Can not transition to error from state " + state());
```

Comment:
Should this be logged at error?

Reply:
In the normal close method the corresponding log is also info. As multiple thread will be calling this at once I would rather not flood the logs with error unnecessarily.

Reply:
Gotcha. In that case maybe we shouldn't log anything here at all? Or just reword it to clarify that this is expected (eg `"Skipping shutdown since we are already in ERROR"`) since "Can not transition..." kind of sounds like something went wrong

Reply:
That is a good idea, Ill change the log